# The ixp4xx_set_irq_type API is missing return statements which causes
# the result to be unusable and the error detection non existent.
#
--- linux-2.6.13/arch/arm/mach-ixp4xx/orig/common.c	2005-09-25 09:52:57.658059680 -0700
+++ linux-2.6.13/arch/arm/mach-ixp4xx/common.c	2005-09-25 09:53:02.354355199 -0700
@@ -125,7 +125,8 @@
 	} else if (type & IRQT_LOW) {
 		int_style = IXP4XX_GPIO_STYLE_ACTIVE_LOW;
 		irq_type = IXP4XX_IRQ_LEVEL;
-	}
+	} else
+		return -EINVAL;
 
 	ixp4xx_config_irq(irq, irq_type);
 
@@ -142,6 +143,8 @@
 
 	/* Set the new style */
 	*int_reg |= (int_style << (line * IXP4XX_GPIO_STYLE_SIZE));
+
+	return 0;
 }
 
 static void ixp4xx_irq_mask(unsigned int irq)
